components/org.wso2.carbon.identity.sts.passive/src/main/java/org/wso2/carbon/identity/sts/passive/utils/RequestProcessorUtil.java

```diff
@@ -202,23 +201,33 @@ public static void addSTSProperties(TokenIssueOperation issueOperation) throws E
         int tenantId = PrivilegedCarbonContext.getThreadLocalCarbonContext().getTenantId();
 
         String[] aliasAndPassword = getKeyStoreAliasAndKeyStorePassword(serverConfig, tenantId, tenantDomain);
+
         String keyAlias = aliasAndPassword[0];
         String keyStorePassword = aliasAndPassword[1];
-        String keyStoreFileLocation = serverConfig
-                .getFirstProperty(RegistryResources.SecurityManagement.SERVER_PRIMARY_KEYSTORE_FILE);
-        String keyStoreName = null;
+        String keyStoreFileLocation = IdentityKeyStoreResolver.getInstance().getKeyStoreConfig(
+                tenantDomain,
+                IdentityKeyStoreResolverConstants.InboundProtocol.WS_FEDERATION,
+                RegistryResources.SecurityManagement.CustomKeyStore.PROP_LOCATION);
+
+        String keyStoreName = generateKSNameFromDomainName(tenantDomain);
 
         String signatureAlgorithm = serverConfig.getFirstProperty(STS_SIGNATURE_ALGORITHM_KEY);
         String digestAlgorithm = serverConfig.getFirstProperty(STS_DIGEST_ALGORITHM_KEY);
 
-
         if (keyAlias == null) {
             throw new STSException("Private key alias cannot be null.");
         }
 
-        if (MultitenantConstants.SUPER_TENANT_ID != tenantId) {
-            keyStoreName = generateKSNameFromDomainName(tenantDomain);
+        // Encryption properties expected by org.apache.wss4j.common.crypto.CryptoFactory is,
+        // If keystore is in <IS-HOME>/repository/resources/security, (Primary keystore or custom keystore)
+        //      keyStoreName = "" or null, keyStoreFileLocation = path to key store
+        // If keystore is not located in file system, (tenant keystore)
+        //      keyStoreName = keystore name, keyStoreFileLocation = "" or any path
+        if (MultitenantConstants.SUPER_TENANT_ID != tenantId && keyStoreFileLocation.equals(keyStoreName)) {
+            keyStoreFileLocation = "";
```

Comment:
Why exactly do we need to set an empty value here? How was this handled with the previous implementation?

Reply:
Shall we move this logic to the `getEncryptionProperties()`? And populate the properties based on the tenant.

Reply:
Made the changes https://github.com/wso2-extensions/identity-inbound-auth-sts/pull/160/commits/32f59928210d4511328f4be380bd21f2375cdf51